module.nix

```diff
@@ -5,49 +5,191 @@ self:
+                };
+                AudioInput = mkOption {
+                    type = types.bool;
+                    default = false;
+                    description = "";
+                };
+                AudioRevealer = mkOption {
+                    type = types.bool;
+                    default = false;
+                    description = "";
+                };
+                AudioScrollSpeed = mkOption {
+                    type = types.nullOr types.int;
+                    default = null;
+                    description = "";
+                };
+                AudioMinVolume = mkOption {
+                    type = types.nullOr types.int;
+                    default = null;
+                    description = "";
+                };
+                AudioMaxVolume = mkOption {
+                    type = types.nullOr types.int;
+                    default = null;
+                    description = "";
+                };
+                NetworkAdapter = mkOption {
+                    type = types.nullOr types.str;
+                    default = null;
+                    description = "";
+                };
+                NetworkWidget = mkOption {
+                    type = types.bool;
+                    default = true;
+                    description = "";
+                };
+                EnableSNI = mkOption {
+                    type = types.bool;
+                    default = true;
+                    description = "";
+                };
+                MinDownloadBytes = mkOption {
+                    type = types.nullOr types.int;
+                    default = null;
+                    description = "";
+                };
+                MaxDownloadBytes = mkOption {
+                    type = types.nullOr types.int;
+                    default = null;
+                    description = "";
+                };
+                MinUploadBytes = mkOption {
+                    type = types.nullOr types.int;
+                    default = null;
+                    description = "";
+                };
+                MaxUploadBytes = mkOption {
+                    type = types.nullOr types.int;
+                    default = null;
+                    description = "";
+                };
+            };
+        };
+    };
+
+    extraConfig = mkOption {
+      type = types.nullOr types.lines;
+      default = '' '';
       description = ''
-        Configuration to write to ~/.config/gBar/config, if none nothing happens
+        Configuration to write to ~/.config/gBar/config, if you want to use your own config then set this to null and home manager wont write anything
       '';
     };
 
-    extraCSS = lib.mkOption {
-      type = lib.types.nullOr lib.types.lines;
+    extraCSS = mkOption {
+      type = types.nullOr types.lines;
       default = null;
       description = ''
         Configuration to write to ~/.config/gBar/config, if none nothing happens
```

Comment:
Shoudn't this be ~/.config/gBar/style.css?
It's probably something that slipped through the last PR.

Reply:
That one's fine, I just chose to not put it under the config submodule, functions at the bottom in config =